doc/specs/#4993 - Keyboard Selection/Keyboard-Selection.md

```diff
@@ -71,6 +76,66 @@ Every combination of the `Direction` and `SelectionExpansionMode` will map to a
 **NOTE**: If `copyOnSelect` is enabled, we need to make sure we update the clipboard on every change in selection.
 
 
+
+### Copy Mode
+Copy Mode is a mode where the user can create and modify a selection using only the keyboard. The following flowchart covers how the new `copyMode` keybinding works:
+
+![Copy Mode Flowchart][images/CopyModeFlowchart.png]
+
+If a selection is not active, a "start" and "end" selection point is created at the cursor position. `moveSelectionPoint` calls then move "start" and "end" together as one position. This position will be denoted with a "y-beam".
+
+![Y-Beam Example][images/Y-Beam.png]
+
+If a selection is active, `copyMode` leaves the selection untouched, all subsequent `moveSelectionPoint` calls move "start". The y-beam is then cut vertically in half, where one half is drawn on the "start" endpoint, and the other half is drawn on the "end" endpoint. Since the user is moving "start", only the half-y-beam is drawn on the "start" endpoint.
+
+![Separated Y-Beam Example][images/Half-Y-Beam.png]
+
+**NOTE:** Both half y-beams could have been presented as shown in the image below. This idea was omitted because then there is no indication for which half y-beam is currently focused.
+
+![Both Separated Y-Beams Example][images/Split-Y-Beam.png]
+
+Pressing `copyMode` again, will then anchor "start". Subsequent `moveSelectionPoint` calls move the "end" selection point. As before, the half-y-beam is only drawn on the "end" to denote that this is the endpoint that is being moved.
+
+Pressing `copyMode` essentially cycles between which selection point is targeted. The half-y-beam is drawn on the targeted endpoint.
+
+#### Block Selection
+A user can normally create a block selection by holding <kbd>alt</kbd> then creating a selection.
+
+If the user is in Copy Mode, and desires to make a block selection, they can use the `toggleBlockSelection` keybinding. `toggleBlockSelection` takes an existing selection, and transforms it into a block selection.
+
+All selections created in Copy Mode will have block selection disabled by default.

```

Comment:
\<some future user\>: can I have an arg to `copyMode` to have it start in block selection mode?

Reply:
Added to future considerations.